onedal/ensemble/forest.py

```diff
@@ -0,0 +1,532 @@
+#===============================================================================
+# Copyright 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+from sklearn.ensemble import BaseEnsemble
+from abc import ABCMeta, abstractmethod
+import numbers
+import warnings
+from sklearn.exceptions import DataConversionWarning
+from sklearn.utils import (compute_sample_weight, check_array, deprecated)
+from sklearn.utils.validation import (_num_samples)
```

Comment:
Minor: single line imports don't need brackets

Reply:
Thank you for the review. Refactoring will be done later.